### DISCUSSION TITLE
Side Buttons Support

### DISCUSSION THREAD

**Discussion by Marcello1099** [linearmouse/linearmouse#12] [Ideas]
The software is fantastic! An additional feature I can think about would be side button support for back and forward.

**Comment by lujjjh** [comment]
Thank you very much for your suggestions!

I'm using a mouse with back and forward buttons, which works perfectly. Are you proposing adding support for customizing mouse buttons (like [SteerMouse](https://plentycom.jp/en/images/product/steermouse/win_main.png))?

**Reply by Marcello1099** [reply]
I was thinking more of something like [SensibleSideButtons](https://sensible-side-buttons.archagon.net) where an incompatible external mouse with side buttons would work. I have seen some that do it through **gestures** or back as _**command + [**_  and forward as _**command + ]**_

**Reply by lujjjh** [reply]
Thanks for the clarification. I think I get what you're talking about. This is a useful feature and I'll add it into the backlog. 

SensibleSideButtons' approach (by generating a gesture) makes more sense. :) However, because the source code is licensed under the GPLv2, I'll have to implement it myself.

**Reply by lujjjh** [reply]
@Marcello1099 I found that linking a GPLv2 library in MIT licenced code is [permissible](https://opensource.stackexchange.com/questions/8134/if-mit-licensed-code-links-to-gpl-licensed-code-what-should-be-the-license-of-th), so I grabbed the [Touch](https://github.com/calftrail/Touch/)'s implementation. Anyway, a universal back and forward is available in [v0.2.7](https://github.com/lujjjh/LinearMouse/releases/tag/v0.2.7). Feel free to have a try.

**Reply by Marcello1099** [reply]
Works great!

**Comment by skareel** [comment]
Please add support for remapping side buttons to other functions. I used steermouse before to map button 4 and button 5 to scrolling up and down in browsers and other programs. I believe you could set how many lines it scrolled too in steermouse so they functionality in here would be amazing

**Reply by lujjjh** [reply]
You can track this in #244.
